core/src/lib.rs

```diff
@@ -130,16 +128,5 @@ pub use transport::Transport;
 pub use upgrade::{InboundUpgrade, OutboundUpgrade, UpgradeError, UpgradeInfo};
 
 #[derive(Debug, thiserror::Error)]
-pub struct DecodeError(String);
-
-impl From<quick_protobuf::Error> for DecodeError {
-    fn from(e: quick_protobuf::Error) -> Self {
-        Self(e.to_string())
-    }
-}
-
-impl fmt::Display for DecodeError {
-    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        write!(f, "{}", self.0)
-    }
-}
+#[error(transparent)]
+pub struct DecodeError(quick_protobuf::Error);
```

Comment:
I am confused. Isn't the constructor of `DecodeError` the same as a `From<quick_protobuf::Error>`? Asked in a different way, changing `DecodeError(quic_protobuf::Error)` to e.g. `DecodeError(SomeNewType)` a breaking change? (Where `SomeNewType` might as well be `quic_protobuf` just as a new version.)

Reply:
Got it. Thanks for explaining!